s3/pkg/meta/redis/redis.go

```diff
@@ -0,0 +1,385 @@
+package redis
+
+import (
+	"strconv"
+
+	"github.com/opensds/multi-cloud/s3/pkg/datastore/yig/config"
+	"github.com/opensds/multi-cloud/s3/pkg/helper"
+)
+
+var (
+	redisClient *RedisCli
+)
+
+const InvalidQueueName = "InvalidQueue"
+
+const keyvalue = "000102030405060708090A0B0C0D0E0FF0E0D0C0B0A090807060504030201000" // This is the key for hash sum !
+
+type RedisDatabase int
+
+func (r RedisDatabase) String() string {
+	return strconv.Itoa(int(r))
+}
+
+func (r RedisDatabase) InvalidQueue() string {
+	return InvalidQueueName + r.String()
+}
+
+const (
+	UserTable RedisDatabase = iota
+	BucketTable
+	ObjectTable
+	FileTable
+	ClusterTable
+)
+
+var MetadataTables = []RedisDatabase{UserTable, BucketTable, ObjectTable, ClusterTable}
+var DataTables = []RedisDatabase{FileTable}
+
+func Initialize(cfg *config.CommonConfig) {
+	redisClient = NewRedisCli()
+	redisClient.Init(cfg)
+}
+
+func Close() {
+	if redisClient != nil && redisClient.IsValid() {
+		err := redisClient.Close()
+		if err != nil {
+			helper.Logger.Printf(2, "Cannot close redis pool, err: %v", err)
+		}
+	}
+}
+
+func GetClient() (*RedisCli, error) {
```

Comment:
a suggestion, can we remove Redis from the nomenclature...the intent is to use a distributed cache here, with Redis being one of them and our current choice